cli/pkg/release/parsing_test.go

```diff
@@ -0,0 +1,206 @@
+			cmdArgs, err := readArgs(tc.args)
+			if err != nil && err.Error() != tc.expectedErrorMsg {
+				t.Fatalf("error messages mismatched, expected %s, received %s", tc.expectedErrorMsg, err.Error())
+			}
+			if err == nil && tc.expectedErrorMsg != "" {
+				t.Fatalf("expected error %v, but no error was raised", tc.expectedErrorMsg)
+			}
+
+			if !cmp.Equal(cmdArgs, tc.expectedCmdArgs, cmp.AllowUnexported(cmdArguments{})) {
+				t.Fatalf("expected args %v, got %v", tc.expectedCmdArgs, cmdArgs)
+			}
+		})
+	}
+}
+
+func TestParseArgs(t *testing.T) {
+	type fileCreation struct {
+		filename string
+		content  string
+	}
+	type testCase struct {
+		setup            []fileCreation
+		name             string
+		cmdArgs          []string
+		expectedParams   *ReleaseParameters
+		expectedErrorMsg string
+	}
+
+	tcs := []testCase{
+		{
+			setup:   []fileCreation{},
+			name:    "no enviornments and manifests",
+			cmdArgs: []string{"--application", "potato"},
+			expectedErrorMsg: "error while reading command line arguments, error: the args --enviornment and --manifest must be set at least once",
+		},
+		{
+			setup: []fileCreation{
+				{
+					filename: "production-manifest.yaml",
+					content: "some production manifest",
+				},
+			},
+			name:    "with environment and manifest",
+			cmdArgs: []string{"--application", "potato", "--environment", "production", "--manifest", "production-manifest.yaml"},
+			expectedParams: &ReleaseParameters{
+				Application: "potato",
+				Manifests: map[string]string{
+					"production": "some production manifest",
+				},
+			},
+		},
+		{
+			setup: []fileCreation{
+				{
+					filename: "development-manifest.yaml",
+					content: "some development manifest",
+				},
+				{
+					filename: "production-manifest.yaml",
+					content: "some production manifest",
+				},
+			},
+			name:    "with environment and manifest multiple times",
+			cmdArgs: []string{"--application", "potato", "--environment", "development", "--manifest", "development-manifest.yaml", "--environment", "production", "--manifest", "production-manifest.yaml"},
+			expectedParams: &ReleaseParameters{
+				Application: "potato",
+				Manifests: map[string]string{
+					"development": "some development manifest",
+					"production": "some production manifest",
+				},
+			},
+		},
+		{
+			name:             "some error occurs in argument parsing",
+			cmdArgs:          []string{"--application"},
+			expectedParams:   nil,
+			expectedErrorMsg: "error while reading command line arguments, error: error while parsing command line arguments, error: flag needs an argument: -application",
+		},
+	}
+
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			// test setup
+			dir, err := os.MkdirTemp("", "kuberpult-cli-test-*")
+			if err != nil {
+				t.Fatalf("error encoutered while creating test directory, error: %v", err)
+			}
+			t.Cleanup(func() {
+				os.RemoveAll(dir)
+			})
+			err = os.Chdir(dir)
```

Comment:
I would prefer not to change dir and instead use absolute paths. Then you can also add `t.parallel()`